### PR TITLE
Adding Metacontroller to charts

### PR DESCRIPTION
**Pull request by willnewby**
#### What this PR does / why we need it:
This PR adds [Metacontroller](https://metacontroller.app/), an addon for creating your own controllers in Kubernetes

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by willnewby** [comment]
/assign @davidkarlsen 

(as recommended by the bot)

**Comment by willnewby** [comment]
Hello?

File: incubator/metacontroller/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{{ if .Values.rbac.create }}
```
Comment (davidkarlsen) [inline incubator/metacontroller/templates/rbac.yaml]: split these into three resource-template files:
- serviceaccount
- role
- rolebinding

File: incubator/metacontroller/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+apiVersion: apps/v1beta2
```
Comment (davidkarlsen) [inline incubator/metacontroller/templates/statefulset.yaml]: use v1

File: incubator/metacontroller/values.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+image:
+  repository: metacontroller/metacontroller
+  tag: 0.2
+
+rbac:
+  create: true
+
+serviceAccount:
+  create: true
+  name: metacontroller
```
Comment (davidkarlsen) [inline incubator/metacontroller/values.yaml]: See https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md for the name and other rbac info

File: incubator/metacontroller/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (davidkarlsen) [inline incubator/metacontroller/templates/rbac.yaml]: use v1

File: incubator/metacontroller/templates/crds.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (davidkarlsen) [inline incubator/metacontroller/templates/crds.yaml]: See https://docs.helm.sh/developing_charts/ for installing CRDs

File: incubator/metacontroller/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
+# Metacontroller
```
Comment (davidkarlsen) [inline incubator/metacontroller/README.md]: aim straight for the stable repo and skip incubating

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/metacontroller/templates/crds.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: compositecontrollers.metacontroller.k8s.io
+  annotations:
```
Comment (willnewby) [inline stable/metacontroller/templates/crds.yaml]: I believe this is the correct way to install these @davidkarlsen, but please let me know if there's some other change I need to make.

**Comment by willnewby** [comment]
Hello?

File: stable/metacontroller/templates/rolebinding.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{{ if .Values.rbac.create }}
+{{ if .Values.serviceAccount.create }}
```
Comment (davidkarlsen) [inline stable/metacontroller/templates/rolebinding.yaml]: this does not depend on serviceAccount.create - only rbac.create, see https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/metacontroller/templates/serviceaccount.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{{ if .Values.rbac.create }}
```
Comment (davidkarlsen) [inline stable/metacontroller/templates/serviceaccount.yaml]: https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md
this only depends on serviceAccount.create

File: stable/metacontroller/README.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+```console
+$ helm install incubator/metacontroller --name my-release \
+  --set=image.tag=0.2,resources.limits.cpu=200m
+```
+
+Alternatively, a YAML file that specifies the values for the above parameters can be provided while installing the chart. For example,
+
+```console
+$ helm install incubator/metacontroller --name my-release -f values.yaml
```
Comment (davidkarlsen) [inline stable/metacontroller/README.md]: stable

File: stable/metacontroller/Chart.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+name: metacontroller
+apiVersion: v1
+appVersion: "0.2"
+description: A Helm chart for Metacontroller
+version: 0.1.0
```
Comment (davidkarlsen) [inline stable/metacontroller/Chart.yaml]: let it be 1.0.0

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

**Comment by willnewby** [comment]
Ready for re-review.

**Review by davidkarlsen — APPROVED** [review]

**Comment by davidkarlsen** [comment]
@willnewby Can you have a look at the failing tests?

**Comment by yinzara** [comment]
Any chance this could be finished?

**Comment by willnewby** [comment]
The chart works and I use it personally, it just doesn't pass the e2e
tests.
---------------
Will Newby
Resident Wizard
Phone: 612.208.3806
Email: willnewby@gmail.com

**Comment by mattfarina** [comment]
The helm hub allows for external charts repos to list their charts. Would be great to get it in there

**Comment by willnewby** [comment]
@mattfarina I'll take a look at getting this added to helm hub. Thanks!

**Comment by luisdavim** [comment]
Is this abandoned?

**Comment by blancsys** [comment]
@willnewby Hi, this is a great work, congratulations. Are you migrating this chart to hub.helm.sh? Where I can find the latest version of this chart? Thanks

**Comment by willnewby** [comment]
@blancsys I put up a pull request to add a repo with it to hub.helm.sh here https://github.com/helm/hub/pull/81

@luisdavim it is not abandoned, I use and maintain it.

**Comment by blancsys** [comment]
@willnewby Great thanks

**Pull request closed without merging**